mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -636,4 +638,120 @@ private ArtifactRepository getArtifactRepository(String runId) {
     URI baseArtifactUri = URI.create(getRun(runId).getInfo().getArtifactUri());
     return artifactRepositoryFactory.getArtifactRepository(baseArtifactUri, runId);
   }
+
+
+  // ********************
+  // * Model Registry *
+  // ********************
+
+  /**
+   * Return the latest model version for each stage.
+   * Currently only the following stages are supported: [None, Staging, Production, Archived].
```

Comment:
1. Update to just call getStages API
2. Add new API getStages

Reply:
I've taken a look at the getModelVersionStages API and for discovering that stages are available, it encounters a chicken-egg problem. You need to know the available stage names but you need to provide a model version to get all the available stage names. The default is to use the latest model version which won't be deployed to any stage.